lightning/backend/sql2kv.go

```diff
@@ -246,6 +348,24 @@ func (kvcodec *tableKVEncoder) Encode(
 		record = append(record, value)
 		kvcodec.tbl.RebaseAutoID(kvcodec.se, value.GetInt64(), false, autoid.RowIDAllocType)
 	}
+
+	if len(kvcodec.genCols) > 0 {
+		mutRow := chunk.MutRowFromDatums(record)
+		for _, gc := range kvcodec.genCols {
+			col := cols[gc.index].ToInfo()
+			evaluated, err := gc.expr.Eval(mutRow.ToRow())
+			if err != nil {
+				return nil, logEvalGenExprFailed(logger, row, col, err)
+			}
+			value, err := table.CastValue(kvcodec.se, evaluated, col, false, false)
+			if err != nil {
+				return nil, logEvalGenExprFailed(logger, row, col, err)
+			}
+			mutRow.SetDatum(gc.index, value)
+			record[gc.index] = value
```

Comment:
LGTM

Why need these two setter both🤔

Reply:
@lance6716 Setting the MutRow is needed for cascading generated columns (think `c as b+1` & `a as b+1`, when calculating `c` we need to calculate `b` first). 
Setting the `record` is for inserting the result into the table. 
`MutRow` and `[]types.Datum` can't be cheaply converted between each other so we keep both of them.